recipe_scrapers/_abstract.py

```diff
@@ -65,41 +66,41 @@ def host(cls) -> str:
         """get the host of the url, so we can use the correct scraper"""
         raise NotImplementedError("This should be implemented.")
 
-    def canonical_url(self):
+    def canonical_url(self) -> str | None:
         canonical_link = self.soup.find("link", {"rel": "canonical", "href": True})
         if canonical_link:
             return urljoin(self.url, canonical_link["href"])
         return self.url
 
-    def title(self):
+    def title(self) -> str | None:
         raise NotImplementedError("This should be implemented.")
 
-    def category(self):
+    def category(self) -> str | None:
         raise NotImplementedError("This should be implemented.")
 
-    def total_time(self):
+    def total_time(self) -> float | None:
         """total time it takes to preparate and cook the recipe in minutes"""
         raise NotImplementedError("This should be implemented.")
 
-    def cook_time(self):
+    def cook_time(self) -> float | None:
         """cook time of the recipe in minutes"""
         raise NotImplementedError("This should be implemented.")
 
-    def prep_time(self):
+    def prep_time(self) -> float | None:
         """preparation time of the recipe in minutes"""
         raise NotImplementedError("This should be implemented.")
 
-    def yields(self):
+    def yields(self) -> str | None:
         """The number of servings or items in the recipe"""
         raise NotImplementedError("This should be implemented.")
 
-    def image(self):
+    def image(self) -> str | None:
         raise NotImplementedError("This should be implemented.")
 
-    def nutrients(self):
+    def nutrients(self) -> dict[str, str | None]:
         raise NotImplementedError("This should be implemented.")
 
-    def language(self):
+    def language(self) -> str | list[str] | None:
```

Comment:
I don't think that this method should return a `list[str]` in any scenario.  If it does, that's probably a bug :/

Reply:
this method from bs4 can return a `list[str]`: https://github.com/sbdchd/recipe-scrapers/blob/c954e1a087e96d5f0c4350ee2f498a366f56056e/recipe_scrapers/rezeptwelt.py#L59-L59

Reply:
I think https://github.com/sbdchd/recipe-scrapers/blob/c954e1a087e96d5f0c4350ee2f498a366f56056e/recipe_scrapers/dr.py#L23-L28 and https://github.com/sbdchd/recipe-scrapers/blob/c954e1a087e96d5f0c4350ee2f498a366f56056e/recipe_scrapers/rezeptwelt.py#L59-L59 can both return `list[str]` according to bs4 types

Reply:
Good find, thanks; that looks like a bug to me.  Raised as #683.